src/pd/Client.cc

```diff
@@ -42,25 +56,33 @@ Client::Client(const std::vector<std::string> & addrs, const ClusterConfig & con
     , config(config_)
     , log(&Logger::get("pingcap.pd"))
 {
-    initClusterID();
-
-    initLeader();
-
-    work_threads_stop = false;
+    init();
+}
 
-    work_thread = std::thread([&]() { leaderLoop(); });
+Client::~Client()
+{
+    work_threads_stop = true;
 
-    check_leader.store(false);
+    if (work_thread.joinable())
+    {
+        work_thread.join();
+    }
 }
 
-Client::~Client()
+void Client::update(const std::vector<std::string> & addrs, const ClusterConfig & config_)
 {
     work_threads_stop = true;
 
     if (work_thread.joinable())
     {
         work_thread.join();
     }
+
+    urls = addrsToUrls(addrs, config_);
+    check_leader = false;
+    config = config_;
```

Comment:
Seems `urls` and `config` can also be moved to `init`. `check_leader` has already existed in `init`.